stac_api/api/extensions/tiles_search.py

```diff
@@ -0,0 +1,231 @@
+"""transaction extension."""
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional, Tuple, Type
+
+import attr
+from cogeo_mosaic.backends.base import BaseBackend
+from cogeo_mosaic.backends.stac import default_stac_accessor
+from cogeo_mosaic.errors import NoAssetFoundError
+from cogeo_mosaic.mosaic import MosaicJSON
+from fastapi import FastAPI, Query
+from morecantile import TileMatrixSet
+from rio_tiler.constants import WEB_MERCATOR_TMS
+from rio_tiler.io import BaseReader, STACReader
+from rio_tiler.models import ImageData
+from rio_tiler.mosaic import mosaic_reader
+from titiler.dependencies import DefaultDependency
+from titiler.endpoints.factory import MosaicTilerFactory
+
+from stac_api.api.extensions.extension import ApiExtension
+from stac_api.clients.postgres.core import CoreCrudClient
+
+
+@attr.s
+class FakeRequest:
```

Comment:
This is to mock the `request` param passed through by default here https://github.com/arturo-ai/arturo-stac-api/blob/7ca0c8790ac64bbd359e88933ef94f25ece5b383/stac_api/api/routes.py#L58-L66

Reply:
A way to get around this is to inject a dependency which itself returns a request object object.  This works because dependencies always have access to the request:

```python
from starlette.requests import Request

def request_dependency(request: Request):
    return request
```